src/notifications/useUnseenCount.ts

```diff
@@ -0,0 +1,39 @@
+import { useEffect, useState } from 'react';
+
+import { useDispatch } from 'react-redux';
+
+import { HeadlessServiceStatus, useHeadlessService } from './useHeadlessService';
+
+import { setUnseenCount } from '@/redux/slices/notifications';
+
+const useInitializeUnseenCount = () => {
+  const { headlessService, status } = useHeadlessService();
+  const [error, setError] = useState<unknown>(null);
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    if (status === HeadlessServiceStatus.READY) {
+      headlessService.fetchUnseenCount({
+        // eslint-disable-next-line @typescript-eslint/no-empty-function
+        listener: () => {},
+        onSuccess: ({ count }) => {
+          dispatch({ type: setUnseenCount.type, payload: count });
+          setError(null);
```

Comment:
Do we need to explicitly set the error to `null` since it's the default status?

Reply:
No, but if there was an error from a previous request, we need to reset it it.